aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1249,6 +1314,39 @@ private boolean s3KeyIdAccessKeyBothConfigured() {
     return (s3AccessKeyId == null) == (s3SecretAccessKey == null);
   }
 
+  private AwsCredentialsProvider credentialsProvider(String credentialsProviderClazz) {
+    try {
+      Class.forName(credentialsProviderClazz);
+      AwsCredentialsProvider provider;
+      if (this.credentialsProviderProperties == null
```

Comment:
this if statement seems unnecessary? because it's equivalent to just 


```
DynMethods.builder("create").impl(credentialsProviderClazz).buildChecked().invoke(credentialsProviderProperties);
```